pac4j-core/src/main/java/org/pac4j/core/client/BaseClient.java

```diff
@@ -95,7 +96,7 @@ public final U getUserProfile(final C credentials, final WebContext context) {
                 }
             }
         }
-        return profile;
+        return Optional.of(profile);
```

Comment:
ofNullable maybe, no?

Reply:
(because can be null at that point.
And actually why `retrieveUserProfile` does not return an `Optional`?

Reply:
In fact I think the Optional should come from `ProfileCreator` and `AuthorizationGenerator`, which are used by `retrieveUserProfile`.